scripts/accounts-invariants/pallet-balances-maintenance.py

```diff
@@ -599,6 +612,150 @@ def set_providers_counter_to_one(chain_connection, account_info, internal_system
+    """
+    Returns contract accounts and code owners.
+    """
+    code_owners = set()
+    contract_accounts = set()
+
+    log.info(f"Querying code owners.")
+    code_info_of_query = chain_connection.query_map('Contracts', 'CodeInfoOf', page_size=1000)
+
+    for (i, (account_id, info)) in enumerate(code_info_of_query):
+        code_owners.add(info.serialize()['owner'])
+        if i % 5000 == 0 and i > 0:
+            log.info(f"Checked {i} code owners")
+    log.info(f"Total code owners is {len(code_owners)}")
+    log.debug(f"Code owners: {code_owners}")
+
+    log.info(f"Querying contract accounts.")
+    contract_info_of_query = chain_connection.query_map('Contracts', 'ContractInfoOf', page_size=1000)
+    for (i, (account_id, info)) in enumerate(contract_info_of_query):
+        contract_accounts.add(account_id.value)
+        if i % 5000 == 0 and i > 0:
+            log.info(f"Checked {i} contracts")
+    log.info(f"Total contracts count is {len(contract_accounts)}")
+    log.debug(f"Contract accounts: {contract_accounts}")
+
+    return code_owners, contract_accounts
+
+
+def no_consumers_zero_reserved(account_id, account_info):
+    """
+    Checks if an account has zero consumers, but non-zero reserved amount.
+    """
+    consumers = account_info['consumers']
+    reserved = account_info['data']['reserved']
+    if consumers == 0 and reserved > 0:
+        log.debug(f"Found an account that has zero consumers and non-zero reserved: {account_id}")
+        return True
+    return False
+
+
+def staker_has_consumers_underflow(account_id, consumers, locks, next_keys):
+    """
+    Checks if an account is a staker or vester that has underflow consumers counter.
+    """
+    vesting_lock_encoded_id = "0x76657374696e6720"
+    staking_lock_encoded_id = "0x7374616b696e6720"
+    if account_id in locks and len(locks[account_id]) > 0:
+        log.debug(f"Account {account_id} has following locks: {locks[account_id]}")
+        vesting_lock = next(filter(lambda lock: lock['id'] == vesting_lock_encoded_id, locks[account_id]), None)
+        if vesting_lock is not None and consumers == 1:
+            log.debug(f"Found vesting lock: {vesting_lock}")
+            log.debug(f"Found an account that has one consumer and vesting lock: {account_id}")
+            return True
+        staking_lock = next(filter(lambda lock: lock['id'] == staking_lock_encoded_id, locks[account_id]), None)
+        if staking_lock is not None:
+            log.debug(f"Found staking lock: {staking_lock}")
+            if consumers == 2:
+                log.debug(f"Found an account that has two consumers and staking lock: {account_id}")
+                return True
+            if consumers == 3 and account_id in next_keys:
+                log.debug(f"Found an account that has three consumers, staking lock, and next session key: "
+                          f"{account_id}")
+                return True
```

Comment:
I know this works even when you implicitely return none but lets add `return False` to the end of this function anyway :)

Reply:
Good point, done